.github/workflows/shisen_ci.yml

```diff
@@ -8,17 +8,21 @@ on:
       - master
       
 jobs:
-  build:
-
+  build-and-deploy:
     runs-on: ubuntu-latest
-    name : Build the Project
-
     steps:
-    - uses: actions/checkout@v2
-    - uses: actions/setup-node@v2
-      with:
-        node-version: '14'
-    - run: npm install
-    - run: npm run build
-    - run: npm test
+      - name: Checkout 
+        uses: actions/checkout@v2
+
+      - name: Install and Build 
+        run: |
+          npm install
+          npm run build
+          npm test
+
+      - name: Deploy
+        uses: JamesIves/github-pages-deploy-action@4.1.0
+        with:
+          branch: gh-pages
+          folder: public 
```

Comment:
Deploy using the `build` folder, as that's the folder which is generated by the `npm run build` command.